src/queue.c

```diff
@@ -6264,8 +6267,15 @@ _dispatch_worker_thread(void *context)
 static unsigned WINAPI
 _dispatch_worker_thread_thunk(LPVOID lpParameter)
 {
-  _dispatch_worker_thread(lpParameter);
-  return 0;
+	HRESULT hr = RoInitialize(RO_INIT_MULTITHREADED);
+
+	_dispatch_worker_thread(lpParameter);
+
+	if (SUCCEEDED(hr)) {
```

Comment:
Perhaps this check should be hoisted?  Do we not want to ensure that the operation was successful?

Reply:
Good point. Does that look right now, or should we also return `hr`?